core/include/irods/private/s3_api/common.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef IRODS_S3_API_ENDPOINT_COMMON_HPP
+#define IRODS_S3_API_ENDPOINT_COMMON_HPP
+
+#include <irods/client_connection.hpp>
+#include <irods/connection_pool.hpp>
+#include <irods/filesystem/object_status.hpp>
+#include <irods/filesystem/permissions.hpp>
+#include <irods/irods_exception.hpp>
+#include <irods/rodsErrorTable.h>
+
+#include <boost/beast/http/status.hpp>
+#include <boost/beast/http/message.hpp>
+#include <boost/beast/http/string_body.hpp>
+#include <boost/url/parse.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <chrono>
+#include <memory>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+struct RcComm;
+
+namespace irods::http
+{
+	class session;
+
+	// clang-format off
+	using field_type    = boost::beast::http::field;
+	using request_type  = boost::beast::http::request<boost::beast::http::string_body>;
+	using response_type = boost::beast::http::response<boost::beast::http::string_body>;
+	using status_type   = boost::beast::http::status;
+	using verb_type     = boost::beast::http::verb;
+
+	using session_pointer_type = std::shared_ptr<irods::http::session>;
+	using request_handler_type = void(*)(session_pointer_type, request_type&);
+
+	using request_handler_map_type = std::unordered_map<std::string_view, request_handler_type>;
+
+	using query_arguments_type = std::unordered_map<std::string, std::string>;
+
+	using handler_type = void (*)(session_pointer_type, request_type&, query_arguments_type&);
+	// clang-format on
+
+	enum class authorization_scheme
+	{
+		basic = 0,
+		openid_connect
+	}; // enum class authorization_scheme
+
+	struct authenticated_client_info
+	{
+		authorization_scheme auth_scheme;
+		std::string username;
+		std::chrono::steady_clock::time_point
+			expires_at; // TODO This may be controlled by OIDC. Think about how to handle that.
```

Comment:
not sure about this OIDC - - note for later - this can probably be removed...

Reply:
Will remove. There may be other references too. Will find them.

Reply:
Done.